### ISSUE TITLE
Parsing float64 numbers behaves differently than encoding/json

### ISSUE THREAD

**Issue by maticz** [valyala/fastjson#22]
Some float64 numbers are parsed differently than when using encoding/json or easyjson.

Here is an example:

```go
package main

import (
	"encoding/json"
	"fmt"

	"github.com/valyala/fastjson"
)

type T struct {
	F float64
}

func main() {
	input := []byte(`{"F": 0.3}`)
	t := &T{}
	json.Unmarshal(input, t)
	fmt.Println("encoding/json:", t.F)

	t = &T{}
	var p fastjson.Parser
	v, _ := p.ParseBytes(input)
	fv := v.Get("F")
	f, _ := fv.Float64()
	fmt.Println("fastjson:     ", f)
}
```

Output:

```
encoding/json: 0.3
fastjson:      0.30000000000000004
```

I know fastjson uses a fast custom implementation of the float64 parser; however, I was just wondering if there is any possibility of getting this in line with how strconv.ParseFloat works.

**Comment by valyala** [comment]
This is a rounding bug in Go - see https://play.golang.org/p/xD5a6sSJLCw .
I'll think on how to work around it...

**Comment by valyala** [comment]
A simple workaround on your side is to use the `%.15g` formatter  - see https://play.golang.org/p/ZnfvVUQsr3Y . It removes the last digit with error in the number.

**Comment by valyala** [comment]
Just discovered that division has no the error:
```go
	n1 := float64(3) * 0.1
	n2 := float64(3) / 10
	fmt.Println(n1, n2)

```

Output:

```
0.30000000000000004 0.3
```

Applied the fix in the commit 85ba5c815204d599d2ed8c8cae4d5d221fd45b42

**Comment by valyala** [comment]
Closing the bug, since parsing `0.3` works fine. Feel free re-opening it if other numbers will be discovered that cannot be properly parsed
